src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/InsDelVariantDetector.java

```diff
@@ -88,4 +90,100 @@ public void inferSvAndWriteVCF(final String vcfOutputFileName, final String samp
+                                                                         final SvDiscoveryDataBundle svDiscoveryDataBundle) {
+
+        final Broadcast<ReferenceMultiSource> referenceBroadcast = svDiscoveryDataBundle.referenceBroadcast;
+        final Broadcast<SAMSequenceDictionary> referenceSequenceDictionaryBroadcast = svDiscoveryDataBundle.referenceSequenceDictionaryBroadcast;
+        final List<SVInterval> assembledIntervals = svDiscoveryDataBundle.assembledIntervals;
+        final Broadcast<SVIntervalTree<VariantContext>> cnvCallsBroadcast = svDiscoveryDataBundle.cnvCallsBroadcast;
+        final StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection discoverStageArgs = svDiscoveryDataBundle.discoverStageArgs;
+        final String sampleId = svDiscoveryDataBundle.sampleId;
+        final Logger toolLogger = svDiscoveryDataBundle.toolLogger;
+
+        final JavaPairRDD<NovelAdjacencyReferenceLocations, Iterable<ChimericAlignment>> narlsAndSources =
+                contigSeqAndChimeras
+                        .flatMapToPair(tigSeqAndChimeras ->
+                                discoverNovelAdjacencyFromChimericAlignments(tigSeqAndChimeras,
+                                        referenceSequenceDictionaryBroadcast.getValue()))   // a filter-passing contig's alignments may or may not produce novel adjacency, hence flatmap
+                        .groupByKey();                                                      // group the same novel adjacency produced by different contigs together
+
+        narlsAndSources.cache();
+
+        SvDiscoveryUtils.evaluateIntervalsAndNarls(assembledIntervals, narlsAndSources, referenceSequenceDictionaryBroadcast,
+                discoverStageArgs, toolLogger);
+
+        List<VariantContext> annotatedVariants =
+                inferTypeAndAnnotateByAlignmentSignatures(narlsAndSources, cnvCallsBroadcast, sampleId,
+                        referenceBroadcast, referenceSequenceDictionaryBroadcast);
+
+        narlsAndSources.unpersist();
+
+        return annotatedVariants;
+    }
+
+    //==================================================================================================================
+
+    /**
+     * Given contig alignments, emit novel adjacency not present on the reference to which the locally-assembled contigs were aligned.
+     */
+    private static Iterator<Tuple2<NovelAdjacencyReferenceLocations, ChimericAlignment>>
+    discoverNovelAdjacencyFromChimericAlignments(final Tuple2<byte[], List<ChimericAlignment>> tigSeqAndChimeras, final SAMSequenceDictionary referenceDictionary) {
+        return Utils.stream(tigSeqAndChimeras._2)
+                .map(ca -> new Tuple2<>(new NovelAdjacencyReferenceLocations(ca, tigSeqAndChimeras._1, referenceDictionary), ca)).iterator();
+    }
+
+
+    // TODO: 1/10/18 to be deprecated
```

Comment:
Can you expand one why and in favor of what in this comment?

Reply:
now moved into method `InsDelVariantDetector.produceVariantsFromSimpleChimeras()`, which is called by both `InsDelVariantDetector.inferSvAndWriteVCF()` and `DiscoverVariantsFromContigAlignmentsSAMSpark.discoverVariantsAndWriteVCF()`.

The two calling path now differ by basically how ChimericAlignment is filtered (one of the main difference between existing logic and new logic) and which contigs are sent to `InsDelVariantDetector.produceVariantsFromSimpleChimeras()` (used to be almost all, even contigs with more than two alignments, now only contigs with exactly two alignments).

Exactly how locations and breakpoint complications are inferred for insertions, deletions and small duplications (ones we assembled across the full alt allele) stay the same.

So in short, for simple variants (inversions are not simple any more), which is the vast majority of calls we emit, the two versions will differ only in preprocessing steps, i.e. which contigs are sent for the simple variation inference.